static/src/js/util/mergeVariables.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Returns a JSON object that is the result of two variable objects being merged
+ * and ignores items with duplicate conceptIds.
+ * @param {object} variables1 - JSON object of a list of variables.
+ * @param {object} variables2 - JSON object of a list of variables.
+ * @return {object} - JSON object containing the merged list variables
+ */
+export const mergeVariables = (variables1, variables2) => {
+  if (variables2.items) { if (!variables1.items) return variables2 }
+  if (!variables2.items) return variables1
+
+  const existingConceptIds = new Set(variables1.items.map((item) => item.conceptId))
```

Comment:
I THINK I may be incorrect here but, https://www.geeksforgeeks.org/lodash-_-merge-method/#
might be doing the same thing if you have the correct argument on the right. We use it in `static/src/js/util/portals.js`